public/main.js

```diff
@@ -0,0 +1,199 @@
+let button = document.getElementById('start')
+let status = document.getElementById('status')
+let squares = Array.from(document.getElementsByClassName('square'))
+let playerX = { name: 'X', moves: [], symbol: 'X' };
+let playerO = { name: 'O', moves: [], symbol: 'O' };
+let currentPlayer;
+let wins = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]];
+let usedMoves = [];
+let timer;
+let count = 0
+let onePlayer = document.getElementById("chooseOne")
+status.textContent = 'Enter names, then press start'
+button.addEventListener('click', gameStart)
+
+function gameStart() {
+    usedMoves = [];
+    let playerOneName = document.getElementById("playerOne").value
+    let playerTwoName = document.getElementById("playerTwo").value
+    playerX = { name: playerOneName || 'Player X', moves: [], symbol: 'X' };
+    playerO = { name: playerTwoName || 'Player O', moves: [], symbol: 'O' };
+    squares.forEach((square) => {
+        square.textContent = ""
+    })
+    currentPlayer = playerX;
+    status.textContent = `${currentPlayer.name}'s Turn`
+    button.disabled = true;
+    gameplay();
+}
+
+function gameplay() {
+    timer = setInterval(countUp, 1000)
+    squares.forEach((square) => { square.addEventListener('click', select) })
+}
+
+function select() {
+    let square = event.target;
+    let i = squares.indexOf(square);
+    if (squares.includes(square) && !usedMoves.includes(parseInt(square.id))) {
+        square.textContent = currentPlayer.symbol;
+        currentPlayer.moves.push(parseInt(squares.slice(i, i + 1)[0].id));
+        usedMoves.push(parseInt(squares.slice(i, i + 1)[0].id));
```

Comment:
You could do this much more simply by just pushing `i` since you've already got that variable and can use it to track the square you don't actually need the square itself.